extensions/mssql/src/credentialstore/nativeCredentialService.ts

```diff
@@ -0,0 +1,145 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AppContext } from '../appContext';
+import { SqlOpsDataClient, ISqlOpsFeature, SqlOpsFeature } from 'dataprotocol-client';
+import * as Utils from '../utils';
+import { ClientCapabilities, RPCMessageType, ServerCapabilities } from 'vscode-languageclient';
+import * as Contracts from './contracts';
+import { Disposable, SecretStorage } from 'vscode';
+import * as azdata from 'azdata';
+import * as UUID from 'vscode-languageclient/lib/utils/uuid';
+
+export class NativeCredentialService extends SqlOpsFeature<any> {
+
+	private static readonly messagesTypes: RPCMessageType[] = [
+		Contracts.DeleteCredentialRequest.type,
+		Contracts.SaveCredentialRequest.type,
+		Contracts.ReadCredentialRequest.type
+	];
+
+	public static asFeature(context: AppContext): ISqlOpsFeature {
+		return class extends NativeCredentialService {
+			private _secretStorage: SecretStorage;
+			private _useNativeCredentialService: boolean;
+			private _passwordsMigrated: boolean;
+
+			constructor(client: SqlOpsDataClient) {
+				super(context, client);
+				this._secretStorage = context.extensionContext.secrets;
+				this._passwordsMigrated = context.extensionContext.globalState.get(Utils.configPasswordsMigrated);
+				this._useNativeCredentialService = Utils.useNativeCredentialsEnabled();
+			}
+
+			override fillClientCapabilities(capabilities: ClientCapabilities): void {
+				Utils.ensure(Utils.ensure(capabilities, 'credentials')!, 'credentials')!.dynamicRegistration = true;
+			}
+
+			override async initialize(capabilities: ServerCapabilities): Promise<void> {
+				this.register(this.messages, {
+					id: UUID.generateUuid(),
+					registerOptions: undefined
+				});
+				if (this._useNativeCredentialService && !this._passwordsMigrated) {
```

Comment:
If the user changes the value of this configuration then this isn't going to happen until they restart, right? Might be good to try and handle that since otherwise users might get confused when it doesn't appear to do anything...

Reply:
@abist You resolved this but I don't see this being handled anywhere. Were you intending on fixing this?

Reply:
(resolving this as I started another thread in the new file...)